csm/core/services/file_transfer.py

```diff
@@ -98,7 +102,12 @@ def save_file(self, dir_to_save, filename, overwrite=False):
                 f'File "{path_to_file_to_save}" already exists. Change ' +
                 '"overwrite" argument if you want to overwrite file')
 
-        copyfile(path_to_cached_file, path_to_file_to_save)
+        try:
+            copyfile(path_to_cached_file, path_to_file_to_save)
+        except PermissionError as pe:
+            Log.warn(f"Incorrect permissions for {path_to_file_to_save}. Changing permissions for USER to RWX: {pe}")
+            os.chmod(dir_to_save, stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
```

Comment:
@kupranay  @soniyamoholkar  @pawankumarsrivastava  
Here I handled user permission. If Permission error occurs while copying the file then change mode operation is performed giving only current csm user RWX permission.

But if the directory  permissions are changed by root user then csm user cannot change the permissions of the directory. In this case Operation Not Permitted error occurs.

Reply:
As per discussion with Pranay and Soniya, Incase of incorrect permission to storage directories, error will be shown on UI indicating incorrect permission